src/app/component/ubs/components/ubs-personal-information/ubs-personal-information.component.ts

```diff
@@ -1,157 +1,206 @@
+        latitude: this.personalData.latitude
+      };
+    } else {
+      address = {
+        id: Date.now(),
+        checked: isChecked,
+        city: newAddress.city,
+        district: newAddress.district,
+        street: newAddress.street,
+        houseCorpus: newAddress.houseCorpus,
+        houseNumber: newAddress.houseNumber,
+        entranceNumber: newAddress.entranceNumber,
+        longitude: newAddress.longitude,
+        latitude: newAddress.latitude
+      };
+    }
+    this.addresses.push(address);
   }
 
-  initForm(): void {
-    this.personalDataForm.setValue({
+  setFormData(): void {
+    this.personalDataForm.patchValue({
       firstName: this.personalData.firstName,
       lastName: this.personalData.lastName,
       email: this.personalData.email,
       phoneNumber: '380' + this.personalData.phoneNumber,
       addressComment: this.personalData.addressComment,
     });
+    this.personalDataForm.addControl('isAddress', new FormControl(this.addresses.length ? 'true' : '', [Validators.required]));
   }
 
-  submit(): void {
-    const personalData: PersonalData = {
-      firstName: this.personalDataForm.get('firstName').value,
-      lastName: this.personalDataForm.get('lastName').value,
-      email: this.personalDataForm.get('email').value,
-      phoneNumber: this.personalDataForm.get('phoneNumber').value.slice(3),
-      city: this.addresses[0].city,
-      district: this.addresses[0].district,
-      street: this.addresses[0].street.split(',').splice(0, 1).join().split(' ').splice(1, 1).join(),
-      houseNumber: this.addresses[0].street.split(',').slice(1).join().trim(),
-      houseCorpus: this.addresses[0].houseCorpus,
-      entranceNumber: this.addresses[0].entranceNumber,
-      addressComment: this.personalDataForm.get('addressComment').value,
-      id: 1,
-      latitude: this.addresses[0].latitude,
-      longitude: this.addresses[0].longitude
+  editAddress(addressId: number) {
+    this.openDialog(true, addressId);
+  }
+
+  addNewAddress() {
+    this.openDialog(false);
+  }
+
+  getControl(control: string) {
+    return this.personalDataForm.get(control);
+  }
+
+  openDialog(isEdit: boolean, addressId?: number): void {
+    const currentAddress = this.addresses.find(address => address.id === addressId);
+    const dialogConfig = new MatDialogConfig();
+    dialogConfig.panelClass = 'address-matDialog-styles';
+    dialogConfig.data = {
+      edit: isEdit,
+      address: isEdit ? currentAddress : {},
     };
-    this.order.personalData = personalData;
-    this.orderService.processOrder(this.order)
-      .pipe(
-        takeUntil(this.destroy)
-      )
-      .subscribe();
-    this.shareFormService.thirdStepObject(this.order);
+    const dialogRef = this.dialog.open(UBSAddAddressPopUpComponent, dialogConfig);
+    dialogRef.afterClosed().pipe(takeUntil(this.destroy)).subscribe((address) => {
+      if (address !== undefined && !isEdit) {
+        this.addAddress(address);
```

Comment:
instead of "address !==undefined" could you check like "address"?

Reply:
fixed